support/documentation/architecture/decisions/0003-use-lombok.md

```diff
@@ -0,0 +1,40 @@
+# Use Lombok
+
+- Status: accepted
+- Date: 2020-05-11
+
+## Context and Problem Statement
+
+Java introduces a lot of boilerplate code that reduces readability of code. How can we reduce that boilerplate code?
+
+## Decision Outcome
+
+We will use Lombok, a library that provides a way to reduce boilerplate code with annotations that generate required functionality. Lombok will be used consistently across the source with only the necessary
+annotations added (thus limiting impact if we decide to remove it later on). We will also define a set of rules for using Lombok to make the code consistent.
+
+Bellow is some standard set of rules for using Lombok inside the library:
+
+- The ordering of annotations should be alphabetically ordering from the class name (e.g. first `@Getter` and then `@Setter`)
+- Lombok's annotations should come last after all the domain specific annotations have been added
+- If only getters and setters are required then Lombok's `@Getter` and `@Setter` annotations should be used (avoid using `@Value`)
+- If the class should be immutable then only `@Getter` and `@RequiredArgsConstructor` annotations should be used
+- If the class contains too many (a decision is left up to the implementor how many is too many) properties then `@Builder` annotations should be considered
+- When constructor is required only `@RequiredArgsConstructor` annotation should be used (unless there is a need of doing some processing in constructor)
+  in which case the standard Java constructor should be used
+- Logging will be done by the Logger generated by `@Slf4J`annotation
+- Other annotations should be added only if required
+- Avoid adding annotations just for the test cases (i.e `@EqualsAndHashCode`)
```

Comment:
Not sure about this one. This can be useful when we want to match some mocks.

Reply:
You should not add functionality to main classes just for tests, you have matchers for these mocks (argThat etc)